src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -326,9 +326,9 @@ else if (exercise instanceof ProgrammingExercise) {
 
         if (exercise != null) {
             List<StudentParticipation> participations = participationService.findByExerciseIdAndStudentIdWithEagerResultsAndSubmissions(exercise.getId(), user.getId());
-
+            List<GradingCriterion> gradingCriteria = gradingCriterionService.findByExerciseIdWithEagerGradingCriteria(exercise.getId());
```

Comment:
I am wondering if this should be fetched in the same call as the exercise above. This would save an extra database call.

Reply:
You're right! They are already fetched above when the exercise is fetched so no need for this database call. I will remove it!

Reply:
I am not sure if this is possible as the criteria are not set by the time the exercise gets fetched above.

Reply:
Well you would need to use something like the `findByIdWithEagerGradingCriteria` method you defined in this PR instead of the service call to just get the Exercise.

Reply:
Ah, I get your point now and changed it accordingly! Thanks for mentioning it.